tree/ntuple/v7/test/ntuple_merger.cxx

```diff
@@ -73,9 +73,360 @@ TEST(RPageStorage, ReadSealedPages)
+      EXPECT_THROW(merger.Merge(sourcePtrs, *destination), ROOT::Experimental::RException);
+   }
+}
+
+TEST(RNTupleMerger, MergeAsymmetric3)
+{
+   // Write two test ntuples to be merged
+   FileRaii fileGuard1("test_ntuple_merge_in_1.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldFoo = model->MakeField<int>("foo", 0);
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard1.GetPath());
+      for (size_t i = 0; i < 10; ++i) {
+         *fieldFoo = i * 123;
+         ntuple->Fill();
+      }
+   }
+
+   FileRaii fileGuard2("test_ntuple_merge_in_2.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldFoo = model->MakeField<int>("foo", 0);
+      auto fieldBar = model->MakeField<int>("bar", 0);
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard2.GetPath());
+      for (size_t i = 0; i < 10; ++i) {
+         *fieldFoo = i * 567;
+         *fieldBar = i * 765;
+         ntuple->Fill();
+      }
+   }
+
+   // Now merge the inputs
+   FileRaii fileGuard3("test_ntuple_merge_out.root");
+   {
+      // Gather the input sources
+      std::vector<std::unique_ptr<RPageSource>> sources;
+      sources.push_back(RPageSource::Create("ntuple", fileGuard1.GetPath(), RNTupleReadOptions()));
+      sources.push_back(RPageSource::Create("ntuple", fileGuard2.GetPath(), RNTupleReadOptions()));
+      std::vector<RPageSource *> sourcePtrs;
+      for (const auto &s : sources) {
+         sourcePtrs.push_back(s.get());
+      }
+
+      // Create the output
+      RNTupleWriteOptions writeOpts;
+      writeOpts.SetUseBufferedWrite(false);
+      auto destination = RPageSink::Create("ntuple", fileGuard3.GetPath(), writeOpts);
+
+      // Now Merge the inputs
+      // We expect this to fail since the fields between the sources do NOT match
+      RNTupleMerger merger;
+      EXPECT_THROW(merger.Merge(sourcePtrs, *destination), ROOT::Experimental::RException);
+   }
+}
+
+TEST(RNTupleMerger, MergeVector)
+{
+   // Write two test ntuples to be merged
+   // These files are practically identical except that filed indices are interchanged
+   FileRaii fileGuard1("test_ntuple_merge_in_1.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldFoo = model->MakeField<std::vector<int>>("foo");
+      auto fieldBar = model->MakeField<std::vector<int>>("bar");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard1.GetPath());
+      for (size_t i = 0; i < 10; ++i) {
+         fieldFoo->clear();
+         fieldBar->clear();
+         fieldFoo->push_back(i * 123);
+         fieldFoo->push_back(i * 456);
+         fieldBar->push_back(i * 789);
+         ntuple->Fill();
+      }
+   }
+
+   FileRaii fileGuard2("test_ntuple_merge_in_2.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldBar = model->MakeField<std::vector<int>>("bar");
+      auto fieldFoo = model->MakeField<std::vector<int>>("foo");
```

Comment:
The `RNTupleMerger` should, in principle, also throw if it detects any mismatch in the type of the field (or any field / column in the field hierarchy).

This should also be validated in a test here, i.e. what would happen if `bar` in `test_ntuple_merge_in_2.root` is not a `std::vector<int>`?

Reply:
Thanks a lot @jalopezg-git. There are at least two things that need to be guaranteed from what I can see, and please correct me if I'm wrong:

When we create the sink from the source's model, the column representatives seem to be copied over as well. Therefore, we need to ensure that the write options that are used for the merged sink (especially compression) mirror the write options that were originally used when the input source was created. Is that information stored in the ntuple itself (i.e. can one get the compression setting of the ntuple from the source)?

Just like the name of the column we can further match the type. Currently I use `RFieldDescriptor` to extract the former and can ensure that `GetTypeName()` and maybe even `GetTypeVersion()` match. Would that suffice?

One more general question. `RFieldDescriptor` seems to only provide the field name relative to the parent when calling `GetFieldName()`. `RField` seems to provide a qualified name `GetQualifiedFieldName()` that also encapsulates the parent information. Is there anything against extending the former to support the latter, too?

Reply:
Regarding the qualified name in the descriptor: That can be queried through the `RNTupleDescriptor::GetQualifiedFieldName()` method. The `RFieldDescriptor` has no back pointer to the `RNTupleDescriptor`, so it cannot resolve the parent ID itself.

I think it would be nice to extend the test to make sure that `std::vector<float>` vs. `std::vector<int>` fails. The current implementation, I think, ensures that.

Reply:
Yes the current implementation will catch inconsistent types. Actually, `MergeInconsistentTypes` already tests that.